parlai/core/metrics.py

```diff
@@ -134,19 +134,19 @@ class Metrics(object):
     def __init__(self, opt):
         self.metrics = {}
         self.metrics['cnt'] = 0
-        self.metrics['correct'] = 0
-        self.metrics['f1'] = 0.0
-        self.custom_metrics = ['mean_rank', 'loss', 'lmloss', 'ppl']
-        for k in self.custom_metrics:
+        self.metrics_list = ['mean_rank', 'loss', 'correct', 'f1']
+        for k in self.metrics_list:
             self.metrics[k] = 0.0
             self.metrics[k + '_cnt'] = 0
         self.eval_pr = [1, 5, 10, 100]
         for k in self.eval_pr:
             self.metrics['hits@' + str(k)] = 0
+        self.metrics['hits@_cnt'] = 0
         if opt.get('numthreads', 1) > 1:
             self.metrics = SharedTable(self.metrics)
 
         self.datatype = opt.get('datatype', 'train')
+        self.has_text_cands = False
```

Comment:
note that these flags aren't shared between threads. we *could* store these in an internal field called "flags" which is a dict or shared table to share them. it looks messier, but if we are adding more then we probably should.

Reply:
@jaseweston thoughts?

Reply:
@jaseweston @alexholdenmiller latest commit creates a SharedTable (in hogwild case) so that the flags are shared between threads if necessary